test/NuGet.Server.Tests/ServerPackageRepositoryTest.cs

```diff
@@ -109,6 +110,41 @@ public void ServerPackageRepositoryRemovePackage()
         }
 
         [Fact]
+        public void ServerPackageRepositoryRetentionRule()
+        {
+            using (var temporaryDirectory = new TemporaryDirectory())
+            {
+                // Arrange
+                Func<string, int, int> getIntSetting = (key, defaultValue) =>
+                {
+                    if (key == "packageRetentionRule")
+                    {
+                        return 3;
+                    }
+                    return defaultValue;
+                };
+
+                var serverRepository = CreateServerPackageRepository(temporaryDirectory.Path, repository =>
+                {
+                    repository.AddPackage(CreatePackage("test", "1.8"));
+                    repository.AddPackage(CreatePackage("test", "1.9"));
+                    repository.AddPackage(CreatePackage("test", "2.0"));
+                    repository.AddPackage(CreatePackage("test", "2.1"));
+                    repository.AddPackage(CreatePackage("test", "2.2"));
+                }, null, getIntSetting);
+
+                // Act
+                serverRepository.AddPackage(CreatePackage("test", "2.3"));
+                var packages = serverRepository.GetPackages().OrderBy(p => p.Version).ToList();
```

Comment:
This has been adjusted to use `FindPackagesById` in the [implementation](https://github.com/NuGet/NuGet.Server/pull/18/files#diff-5fe4d2ce57f43bc716ec5562ea2063f8R271). The test should follow the same code path and use `FindPackagesById` too.